lib/compat/wordpress-6.8/class-gutenberg-rest-post-counts-controller.php

```diff
@@ -0,0 +1,217 @@
+<?php
+/**
+ * REST API: Gutenberg_REST_Post_Counts_Controller class
+ *
+ * @since 6.8.0
+ *
+ * @package gutenberg
+ * @subpackage REST_API
+ */
+
+/**
+ * Core class used to return post counts by post type via the REST API.
+ *
+ * @since 6.8.0
+ *
+ * @see WP_REST_Controller
+ */
+class Gutenberg_REST_Post_Counts_Controller extends WP_REST_Controller {
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		$this->namespace = 'wp/v2';
+		$this->rest_base = 'counts';
+	}
+
+	/**
+	 * Registers the routes for post counts.
+	 *
+	 * @since 6.8.0
+	 *
+	 * @see register_rest_route()
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/(?P<post_type>[\w-]+)',
+			array(
+				'args'   => array(
+					'post_type' => array(
+						'description' => __( 'An alphanumeric identifier for the post type.' ),
+						'type'        => 'string',
+					),
+				),
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_item' ),
+					'permission_callback' => array( $this, 'get_item_permissions_check' ),
+					'args'                => array(
+						'context' => $this->get_context_param( array( 'default' => 'view' ) ),
+					),
+				),
+				'schema' => array( $this, 'get_public_item_schema' ),
+			)
+		);
+	}
+
+	/**
+	 * Checks if a given request has access to read post counts.
+	 *
+	 * @since 6.8.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return true|WP_Error True if the request has read access, WP_Error object otherwise.
+	 */
+	public function get_item_permissions_check( $request ) {
+		$post_type = get_post_type_object( $request['post_type'] );
+
+		if ( ! $post_type ) {
+			return new WP_Error(
+				'rest_invalid_post_type',
+				__( 'Invalid post type.' ),
+				array( 'status' => 404 )
+			);
+		}
+
+		if ( ! current_user_can( $post_type->cap->edit_posts ) ) {
+			return new WP_Error(
+				'rest_cannot_read',
+				__( 'Sorry, you are not allowed to read post counts for this post type.' ),
+				array( 'status' => rest_authorization_required_code() )
+			);
+		}
+
+		return true;
+	}
+
+	/**
+	 * Retrieves post counts for a specific post type.
+	 *
+	 * @since 6.8.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function get_item( $request ) {
+		$post_type = $request['post_type'];
+		$counts    = wp_count_posts( $post_type );
+
+		if ( ! $counts ) {
```

Comment:
`wp_count_posts()` always seems to return a `stdClass` object, so it’s unlikely to ever trigger this 500 error, as the `$counts` variable will always evaluate to `true`.

Reply:
Good point, thanks for catching.

`get_item_permissions_check` will already do an `rest_invalid_post_type` type check anyway, so I'll remove this block.